dart/dynamics/InverseKinematics.h

```diff
@@ -492,6 +573,250 @@ class InverseKinematics : public common::Subject
                          Eigen::VectorXd& _grad) override;
   };
 
+  /// Analytical is a base class that should be inherited by methods that are
+  /// made to solve the IK analytically instead of iteratively. This provides an
+  /// extended API that is relevant to Analytical solvers but not iterative
+  /// solvers.
+  ///
+  /// Creating an Analytical solver will have the side effect of removing the
+  /// error clamp and error weights from your ErrorMethod. If you still want
+  /// your error computations to be clamped and weighted, you should set it
+  /// again after creating the Analytical solver.
```

Comment:
I think it would be good to mention the reason of removing the error clamp and error weights since we have a good reason. The reason seems to be well described in `InverseKinematics::clone(JacobianNode*)`. :smile: